File: src/store.js
/**
 * store.js - 精简持久化模块
 * 用单个 JSON 文件保存「设置」与「下载任务」，替代原项目里的 sql.js 数据库 + electron-store，
 * 减少依赖，方便独立打包。
 */
const fs = require('fs');
const path = require('path');

let dataFile = null;
let cache = null; // { settings, tasks }

function loadCache() {
  if (!dataFile) throw new Error('store 未初始化');
  if (cache) return cache;
  try {
    if (fs.existsSync(dataFile)) {
      cache = JSON.parse(fs.readFileSync(dataFile, 'utf8'));
    }
  } catch (e) {
    console.error('[Store] 读取数据文件失败，使用空数据:', e.message);
  }
  if (!cache || typeof cache !== 'object') cache = {};
  if (!Array.isArray(cache.tasks)) cache.tasks = [];
  if (!cache.settings || typeof cache.settings !== 'object') cache.settings = {};
  return cache;
}

function flush() {
  if (!dataFile) return;
  try {
    fs.mkdirSync(path.dirname(dataFile), { recursive: true });
    fs.writeFileSync(dataFile, JSON.stringify(cache || {}, null, 2), 'utf8');
  } catch (e) {
    console.error('[Store] 写入数据文件失败:', e.message);
  }
}

function init(filePath) {
  dataFile = filePath;
  loadCache();
}

function getSettings() {
  return loadCache().settings;
}

function saveSettings(settings) {
  loadCache().settings = settings || {};
  flush();
}

function getTasks() {
  return loadCache().tasks;
}

function saveTasks(tasks) {
  loadCache().tasks = tasks || [];
  flush();
}

module.exports = {
  init,
  getSettings,
  saveSettings,
  getTasks,
  saveTasks,
};

File: store.py
"""
store.py - 精简持久化模块（Python 版）

用单个 JSON 文件保存「设置」与「下载任务」，减少依赖，方便 Web/Docker 部署。
对应原 Node 实现：src/store.js
"""
import json
import os

data_file = None
cache = None


def _load_cache():
    global cache
    if not data_file:
        raise Exception("store 未初始化")
    if cache is not None:
        return cache
    try:
        if os.path.exists(data_file):
            with open(data_file, "r", encoding="utf-8") as f:
                cache = json.load(f)
    except Exception as e:
        print("[Store] 读取数据文件失败，使用空数据:", e)
    if not isinstance(cache, dict):
        cache = {}
    if not isinstance(cache.get("tasks"), list):
        cache["tasks"] = []
    if not isinstance(cache.get("settings"), dict):
        cache["settings"] = {}
    return cache


def _flush():
    if not data_file:
        return
    try:
        os.makedirs(os.path.dirname(data_file), exist_ok=True)
        with open(data_file, "w", encoding="utf-8") as f:
            json.dump(cache or {}, f, ensure_ascii=False, indent=2)
    except Exception as e:
        print("[Store] 写入数据文件失败:", e)


def init(file_path):
    global data_file
    data_file = file_path
    _load_cache()


def get_settings():
    return _load_cache().get("settings")


def save_settings(settings):
    _load_cache()["settings"] = settings or {}
    _flush()


def get_tasks():
    return _load_cache().get("tasks")


def save_tasks(tasks):
    _load_cache()["tasks"] = tasks or []
    _flush()
